go/private/actions/archive.bzl

```diff
@@ -171,19 +167,19 @@ def emit_archive(go, source = None, _recompile_suffix = "", recompile_internal_d
         pathtype = source.library.pathtype,
 
         # GoSource fields
-        srcs = as_tuple(source.srcs),
-        orig_srcs = as_tuple(source.orig_srcs),
-        _cover = as_tuple(source.cover),
-        _embedsrcs = as_tuple(source.embedsrcs),
+        srcs = tuple(source.srcs),
```

Comment:
Do we even need these `tuple` calls? Provider contents are frozen when the target has been evaluated, so we wouldn't need to worry about immutability.

Reply:
hmm are you sure? Here's what I see without `tuple`:
```
Traceback (most recent call last):
	File "/Users/dzbarsky/rules_go/go/private/rules/library.bzl", line 40, column 25, in _go_library_impl
		archive = go.archive(go, source)
	File "/Users/dzbarsky/rules_go/go/private/actions/archive.bzl", line 212, column 28, in emit_archive
		transitive = depset([data], transitive = [a.transitive for a in direct]),
Error in depset: depset elements must not be mutable values
```

Reply:
Right, didn't think of `depset`. The Java rules use complex logic where they join all args into an escaped string and then store that in a depset only to tokenize it later but I would prefer not to do that here.